src/software/ai/navigator/navigator.cpp

```diff
@@ -2,303 +2,324 @@
-                        this->current_robot->velocity().len(),
-                        Util::DynamicParameters->getNavigatorConfig()
-                            ->RobotObstacleInflationFactor()
-                            ->value(),
-                        Util::DynamicParameters->getNavigatorConfig()
-                            ->VelocityObstacleInflationFactor()
-                            ->value()));
-            }
-
-            this->current_robot = std::nullopt;
+            assigned_primitives.emplace_back(std::move(current_primitive));
         }
-        assigned_primitives.emplace_back(std::move(current_primitive));
     }
 
+    Rectangle navigable_area(
+        Point(this->world.field().totalXLength(), this->world.field().totalYLength()),
+        this->world.field().totalXLength(), this->world.field().totalYLength());
+
+    auto paths = path_manager->getManagedPaths(path_objectives, navigable_area,
+                                               getNonPathPlanningObstacles());
+    addPathsToPrimitives(paths, assigned_primitives);
+
     return assigned_primitives;
 }
 
-std::vector<Obstacle> Navigator::createCurrentObstacles(
-    const std::vector<AvoidArea> &avoid_areas, unsigned int robot_id)
+void Navigator::addPathsToPrimitives(
+    const std::map<int, Path> &paths,
+    std::vector<std::unique_ptr<Primitive>> &assigned_primitives)
 {
-    std::vector<Obstacle> obstacles = velocity_obstacles;
-
-    // Avoid obstacles specific to this MoveIntent
-    for (auto area : avoid_areas)
+    for (const auto &path : paths)
     {
-        if (area == AvoidArea::ENEMY_ROBOTS)
+        // look for move intent
+        auto intent_it = path_planning_intents.find(path.first);
+        if (intent_it == path_planning_intents.end())
         {
-            for (auto &robot : world.enemyTeam().getAllRobots())
+            std::stringstream ss;
+            ss << "Tried to find intent associated with robot id = " << path.first
+               << ", but failed";
+            throw std::runtime_error(ss.str());
+        }
+        MoveIntent intent = intent_it->second;
+        if (path.second)
+        {
+            std::vector<Point> path_points = path.second->getKnots();
+            planned_paths.emplace_back(path_points);
+
+            if (path_points.size() == 1)
             {
-                Obstacle o = Obstacle::createRobotObstacleWithScalingParams(
-                    robot,
-                    Util::DynamicParameters->getNavigatorConfig()
-                        ->RobotObstacleInflationFactor()
-                        ->value(),
-                    Util::DynamicParameters->getNavigatorConfig()
-                        ->VelocityObstacleInflationFactor()
-                        ->value());
-                obstacles.push_back(o);
+                throw std::runtime_error(
+                    "Path only contains one point, which is invalid, since it's ambiguous if it's the start or end or some other point");
```

Comment:
done

Reply:
Looks like this wasn't changed?

Reply:
Whoops. Yeah do we assume no path here? I think that's too gentle for what's a pretty bad bug if it returns with one point

Reply:
Hmm. Honestly I'd rather just treat a single point as both the start and end (@MathewMacDougall for your opinion on this). If we're going to treat it as invalid I'd probably just send a `StopPrimitive` to that robot and log a warning? It's definitely a bad bug, but I don't think it's indicative that the AI is in an unrecoverable state?

Reply:
@MathewMacDougall ?

Reply:
We will move to the single point with the implicit assumption that the single point represents the start and destination (because of the guarantee that a path contains start and dest)